crates/blockchain-tree/src/bundle.rs

```diff
@@ -18,7 +18,7 @@ pub struct BundleStateDataRef<'a> {
 }
 
 impl<'a> BundleStateDataProvider for BundleStateDataRef<'a> {
-    fn state(&self) -> &BundleStateWithReceipts {
+    fn state(&self) -> &BlockExecutionOutcome {
```

Comment:
hmm, while naming is an improvement, this does not fit if we keep using it like this.

Reply:
Yes I thought we were saving it for another PR as there were a lot of changes to make but it's true that it doesn't make much sense to merge like that.

So I tried changing the names around `BlockExecutionOutcome` for overall consistency, hope I didn't forget any.